src/mpi/init/globals.c

```diff
@@ -15,3 +15,12 @@ MPIR_Thread_info_t MPIR_ThreadInfo;
    MPI_Init and MPI_Finalize) */
 MPIU_DLL_SPEC MPI_Fint *MPI_F_STATUS_IGNORE MPL_USED = 0;
 MPIU_DLL_SPEC MPI_Fint *MPI_F_STATUSES_IGNORE MPL_USED = 0;
+
+void (**MPIR_QMPI_pointers) (void);
+void **MPIR_QMPI_storage;
+void (**MPIR_QMPI_tool_init_callbacks) (int);
+int MPIR_QMPI_num_tools = 0;
+char **MPIR_QMPI_tool_names;
+int MPIR_QMPI_is_initialized = 0;
+void (**MPIR_QMPI_first_fn_ptrs) (void);
+int *MPIR_QMPI_first_tool_ids;
```

Comment:
Why they are not in `src/mpi_t/qmpi_register.c`?

Reply:
Fine with me.